utils/sync/util.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2017 Pantheon technologies s.r.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//Package sync contains a set of counters synchronizer related utilities
+package sync
+
+import (
+	"errors"
+	"fmt"
+	log "github.com/ligato/cn-infra/logging/logrus"
+	"sync/atomic"
+	"time"
+)
+
+//WaitCounterMatch waits until counter matches
+//an expected number during an specific time.
+func WaitCounterMatch(duration time.Duration, expected uint32, counter *uint32) error {
```

Comment:
Let's use core.EventLoopWithInterrupt() instead of having this awkward function.